File: AfantazieServer/Afantazie.Presentation.Model/Dto/NotificationDto.cs
using Afantazie.Core.Model;
using System;
using System.Collections.Generic;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Afantazie.Presentation.Model.Dto
{
    public class NotificationDto
    {
        public int Id { get; set; }
        public byte Type { get; set; } = (byte)NotificationType.Reply;

        public string Color { get; set; } = "";

        public string DateCreated { get; set; } = "";

        public int? ThoughtId { get; set; } = null;

        public string? ThoughtTitle { get; set; } = "";

        public string? ThoughtAuthor { get; set; } = "";

        public bool isRead { get; set; } = false;
    }
}

File: AfantazieWeb/src/api/dto/NotificationDto.ts
export interface notificationDto {
    id: number;
    dateCreated: string;
    thoughtTitle: string | undefined;
    thoughtId: number | undefined;
    thoughtAuthor: string | undefined;
    isRead: boolean;
    color: string;
    type: NotificationType;
}

export enum NotificationType {
    NewReply = 0,
    Announcement = 100,
    SystemMessage = 101,
}